src/lib389/lib389/schema.py

```diff
@@ -116,15 +116,51 @@ def get_attr_syntaxes(json=False):
             result = ATTR_SYNTAXES
         return result
 
-    def _get_schema_objects(self, object_model, json=False):
-        """Get all the schema objects for a specific model: Attribute, Objectclass,
-        or Matchingreule.
+    def gather_oc_sup_attrs(self, oc, sup_oc, ocs):
+        """
+        Recursively build up all the objectclass superiors' may/must
+        attributes
+        """
+        if sup_oc is None:
+            # First objectclass is none
+            sup_oc = oc
+        for soc in sup_oc['sup']:
+            if soc.lower() == "top":
+                continue
+            # Get sup_oc
+            for obj in ocs:
+                oc_dict = vars(ObjectClass(obj))
+                name = oc_dict['names'][0]
+                if name.lower() == soc.lower():
+                    # Found the superior, get it's attributes
+                    for attr in oc_dict['may']:
+                        if attr not in oc['may']:
+                            oc['may'] = oc['may'] + (attr,)
+                    for attr in oc_dict['must']:
+                        if attr not in oc['must']:
+                            oc['must'] = oc['must'] + (attr,)
+
+                    # Sort the tuples
+                    may = list(oc['may'])
+                    may.sort()
+                    oc['may'] = tuple(may)
```

Comment:
Minor neat pick, but I think it'll be a bit cleaner to use `sorted()` directly instead of `list() -> .sort()` conversion.
And it can be done with one line - `oc['may'] = tuple(sorted(oc['may']))`